### PR TITLE
[ARTP-184] Expand theme type definition

### PR DESCRIPTION
**Pull request by evanrs**

### CODE REVIEW

File: src/client/src/rt-theme/themes.ts [hunk 1]
```diff
@@ -109,15 +153,4 @@ export function createTheme({ primary, secondary }: PaletteMap, accents: Palette
   return _.merge(theme, overrides)
```
Comment (bhavesh-desai-scratch) [inline src/client/src/rt-theme/themes.ts]: Does a spread no work here?

File: src/client/src/rt-theme/themes.ts [hunk 2]
```diff
@@ -1,53 +1,100 @@
...
+  shell: ColorPair
+  component: ColorPair
+
+  button: TouchableIntents
+
+  // Known extensible properties
+  backgroundColor?: Color
+  textColor?: Color
+
+  // TODO: remove after theme migration
```
Comment (bhavesh-desai-scratch) [inline src/client/src/rt-theme/themes.ts]: Can we date and initial any checked in TODO's plz

Comment (evanrs) [reply src/client/src/rt-theme/themes.ts]: How has this never occurred to me?? 

Yes, definitely.

File: src/client/src/rt-theme/themes.ts (hunk 1 above)
Comment (evanrs) [reply src/client/src/rt-theme/themes.ts]: Will replace with a customizer fn.

**Pull request merged**